analyses/malawi/flooding_trigger/03_combine_events.md

```diff
@@ -0,0 +1,173 @@
+    dict_emdat[station] = df_sel
+    if SAVE_DATA: df_sel.to_csv((EXPLORE_DIR / f'{station}_emdat_event_summary.csv'), index=False)
+```
+
+We'll also manually add in the 2015 CERF activation. The exact location is still unclear, but for now we'll assume that it was in both districts. The start and end dates are also somewhat unclear. I've used the date of Govt emergency declaration as the start date, and the date of the Preliminary Flood Response Plan as the end date.
+
+```python
+dict_cerf = {}
+dates = [['2015-01-13', '2015-01-22']]
+
+for station in stations_adm2.values():
+    dict_cerf[station] = pd.DataFrame(dates, columns=['start_date', 'end_date'])
+```
+
+Let's tag each event by its source and combine them into a single dataframe for each station.
+
+```python
+dict_combined = {}
+
+for station in stations_adm2.values():
+    dict_floodscan[station]['source'] = 'floodscan'
+    dict_rco[station]['source'] = 'rco'
+    dict_emdat[station]['source'] = 'emdat'
+    dict_cerf[station]['source'] = 'cerf'
+    
+    df1 = dict_floodscan[station].append(dict_rco[station], ignore_index=True)
+    df2 = df1.append(dict_emdat[station], ignore_index=True)
+    df3 = df2.append(dict_cerf[station], ignore_index=True)
+    
+    df3['start_date'] = pd.to_datetime(df3['start_date'])
+    df3['end_date'] = pd.to_datetime(df3['end_date'])
+    
+    dict_combined[station] = df3[['start_date', 'end_date', 'source']].sort_values(by='start_date')
+    
+    # Save this out to a csv
+    if SAVE_DATA: dict_combined[station].to_csv(EXPLORE_DIR / f'{station}_all_events.csv', index=False)
+```
+
+Create a new dataframe to keep track of overlapping events. 
+
+```python
+all_dates = pd.date_range('1998-01-01', '2019-12-31')
+sources = dict_combined['Nsanje']['source'].unique()
+
+buffer = 10
+
+dict_event_daily = {}
+
+for station in stations_adm2.values(): 
+    df_event_daily = pd.DataFrame(0, index=np.arange(len(all_dates)), columns=sources)
+    df_event_daily['date'] = all_dates
+
+    for source in sources: 
+        df_sel = dict_combined[station]
+        df_sel = df_sel[df_sel['source']==source]
+
+        for index, row in df_sel.iterrows():
+            event_start = pd.to_datetime(df_sel.loc[index, 'start_date']) - timedelta(days=buffer)
+            event_end = pd.to_datetime(df_sel.loc[index, 'end_date']) + timedelta(days=buffer)
+            df_event_daily[source] = np.where((df_event_daily['date'] <= event_end) & (df_event_daily['date'] >= event_start), 1, df_event_daily[source])
+
+    df_event_daily['source_sum'] = df_event_daily.sum(numeric_only = True, axis=1)
+    
+    dict_event_daily[station] = df_event_daily
+
+    fig, ax = plt.subplots(figsize=(10, 3))
+    ax.set_title(station)
+    ax.plot(df_event_daily['date'], df_event_daily['source_sum'], '-', label=station)
+    ax.set_xlabel('Date')
+    ax.set_ylabel('Number of sources confirming event')
+    
+    if SAVE_PLOT: plt.savefig(PLOT_DIR / f'{station}_event_summary.png')
+```
```

Comment:
really nice! Would also be interesting to see which overlap often then (as it seems some sources just have events more often than others as well?), but I know that is outside the scope ;) Well done :)